lifecycle-operator/controllers/lifecycle/keptnworkloadversion/controller.go

```diff
@@ -152,43 +140,115 @@ func (r *KeptnWorkloadVersionReconciler) Reconcile(ctx context.Context, req ctrl
 	}
 
 	// Wait for deployment of Workload
-	phase = apicommon.PhaseWorkloadDeployment
+	result, err = r.doDeploymentPhase(ctx, workloadVersion, phaseHandler, ctxWorkloadTrace)
+	if !result.Continue {
+		return result.Result, err
+	}
+
+	// Wait for post-deployment checks of Workload
+	result, err = r.doPostDeploymentTaskPhase(ctx, workloadVersion, phaseHandler, ctxWorkloadTrace)
+	if !result.Continue {
+		return result.Result, err
+	}
+
+	// Wait for post-evaluation checks of Workload
+	result, err = r.doPostDeploymentEvaluationPhase(ctx, workloadVersion, phaseHandler, ctxWorkloadTrace)
+	if !result.Continue {
+		return result.Result, err
+	}
+
+	// WorkloadVersion is completed at this place
+	return r.finishKeptnWorkloadVersionReconcile(ctx, workloadVersion, spanWorkloadTrace)
+}
+
+func (r *KeptnWorkloadVersionReconciler) doPreDeploymentTaskPhase(ctx context.Context, workloadVersion *klcv1alpha4.KeptnWorkloadVersion, phaseHandler controllercommon.PhaseHandler, ctxWorkloadTrace context.Context) (*controllercommon.PhaseResult, error) {
+	if !workloadVersion.IsPreDeploymentSucceeded() {
+		reconcilePre := func(phaseCtx context.Context) (apicommon.KeptnState, error) {
+			return r.reconcilePrePostDeployment(ctx, phaseCtx, workloadVersion, apicommon.PreDeploymentCheckType)
+		}
+		return phaseHandler.HandlePhase(ctx,
+			ctxWorkloadTrace,
+			r.getTracer(),
+			workloadVersion,
+			apicommon.PhaseWorkloadPreDeployment,
+			reconcilePre,
+		)
+	}
+	return &controllercommon.PhaseResult{
```

Comment:
we could make `&controllercommon.PhaseResult{Continue: true}` a constant so we don't need to allocate these bytes multiple time

Reply:
I would keep this as is for now, as the PhaseHandler returns a `PhaseResult` pointer, which means that if we introduce a global variable for this particular value, we would return the pointer to to this variable here. This might be dangerous, as Go does not support constant structs, which means if the properties of this variable are changed accidentally, this will affect all places where this variable is used.

Reply:
true, it is a pointer 😭  I am fine with this!